docs/dev-guide/contributing/dev_setup.rst

```diff
@@ -5,12 +5,90 @@ Developer Setup
+is the easiest way to get started on developing with Pulp if you aren't sure which method you
+prefer. Vagrant is available in Fedora 21 and newer. Follow these steps:
+
+#. Install vagrant::
+   
+   $ sudo yum install vagrant-libvirt
+
+#. Install NFS. NFS will be used to share your code directory with the deployed virtual machine::
+   
+   $ sudo yum install nfs-utils
+
+#. You will need to grant the nfsnobody user rx access to the folder that you check out your code
+   under. Many developers check out code into $HOME/devel or similar. In Fedora, $HOME typically
+   does not allow such access to other users. If your code is in $HOME, you will need to::
+   
+   $ setfacl -m user:nfsnobody:r-x $HOME  # Season to taste, as per above
+
+#. Start and enable the nfs-server service::
+
+   $ sudo systemctl enable nfs-server && sudo systemctl start nfs-server
+
+#. You will need to allow NFS services through your firewall::
+   
+   $ sudo firewall-cmd --permanent --add-service=nfs
+   $ sudo firewall-cmd --permanent --add-service=rpc-bind
+   $ sudo firewall-cmd --permanent --add-service=mountd
+   $ sudo firewall-cmd --reload
+
+#. You are now prepared to check out the Pulp code into your preferred location. Change directories
+   to that location, and check out at least the platform and the RPM plugins. The RPM plugins are
+   needed as the provisioning script configures and synchronizes an example repository::
+
+   $ cd $HOME/devel  # Season to taste
+   $ git clone git@github.com:pulp/pulp.git && git clone git@github.com:pulp/pulp_rpm.git
+
+#. Optionally, check out some of our other plugins as well::
+
+   $ git clone git@github.com:pulp/pulp_docker
+   $ git clone git@github.com:pulp/pulp_python
+   $ git clone git@github.com:pulp/pulp_puppet
+   $ git clone git@github.com:pulp/pulp_ostree
+
+#. Next, cd into the pulp directory, and begin provisioning your Vagrant environment. Due to what
+   seems to be a `Vagrant bug <https://github.com/mitchellh/vagrant/issues/5321>`_, the first run
+   will fail due to Vagrant attempting to control NFS through init scripts rather than through
+   systemd. There is an easy workaround, as you can tell Vagrant to reload the box after it
+   configures your NFS exports, and everything should work. Another possible failure point is during
+   provisioning when mongod is building its initial files. This sometimes takes longer than systemd
+   allows and can fail. If that happens, simple run ``vagrant provision``. We will run two vagrant
+   reloads in a row. The first is a workaround for the mentioned bug, and the second allows the
+   machine to reboot after provisioning.::
+
+   $ cd pulp
+   $ vagrant up  # You will experience an error starting nfs and rpcbind here
+   $ vagrant reload  # mongod may fail when this runs. vagrant provision will fix it.
+   $ vagrant reload  # Reboot the machine at the end to apply kernel updates, etc.
```

Comment:
On 04/23/2015 07:48 PM, Chris Duryee wrote:
> |mount.nfs: access denied by server while mounting 192.168.121.1:/path/to/my/code/dir|

This is the error message I got before I learned that I needed to grant
r-x access to the nfsnobody user to my home dir. You might need to give
that user access to one of the parent dirs (i.e., if /path/to is 700,
nfsnobody will need a facl to /path/to in order to get to the full
path). Do you think that is the problem, or something else?

Another possibility is that it's selinux related. I didn't need to
adjust any selinux rules in order to serve /home/rbarlow/devel, but
perhaps if you are trying with a different path you will need special
SELinux contexts?

-- 
Randy Barlow